### PR TITLE
GEODE-9132: Cleanup PartitionedRegionClear

### PR DESCRIPTION
**Pull request by kirklund**
```
commit 425f25b3e0fbcd8681dc0f76340281b1ec143798
Author: Kirk Lund <klund@apache.org>
Date:   Fri Apr 9 17:24:26 2021 -0700

    GEODE-9132: Cleanup PartitionedRegionClearMessage
    
    * Use descriptive names for variables and methods
    * Use Objects.requireNonNull instead of Assert.assertTrue
    * Remove unnecessary uses of final, this, and super
    * Use static logger
    * Reformat some lines with weird formatting
```
```
commit a4fa93d30819924bd1ec291470c60743e950f89f
Author: Kirk Lund <klund@apache.org>
Date:   Fri Apr 9 15:22:46 2021 -0700

    GEODE-9132: Fixup PartitionResponse constructors
    
    * Chain constructors to only one constructor that calls super
    * Expose all arguments for dependency injection
    * Provide type to recipients
```
```
commit a2c41bf9a2bcb1b42d5fe6d85218b9ded8291ce1
Author: Kirk Lund <klund@apache.org>
Date:   Fri Apr 9 15:06:40 2021 -0700

    GEODE-9132: Delete ClearPRMessage
```

### CODE REVIEW

**Review by DonalEvans — APPROVED** [review]

**Review by dschneider-pivotal — APPROVED** [review]

**Review by agingade — APPROVED** [review]

**Review by Bill — CHANGES_REQUESTED** [review]
A few issues below…

File: geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegionClear.java [hunk 1]
```diff
@@ -0,0 +1,506 @@
...
+  void acquireDistributedClearLock(String clearLock) {
+    try {
+      partitionedRegion.getPartitionedRegionLockService().lock(clearLock, -1, -1);
+    } catch (IllegalStateException e) {
+      partitionedRegion.lockCheckReadiness();
+      throw e;
+    }
+  }
+
+  void releaseDistributedClearLock(String clearLock) {
```
Comment (mhansonp) [inline geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegionClear.java:75]: Isn't clear lock just an object?

File: geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegionClear.java [hunk 2]
```diff
@@ -0,0 +1,506 @@
...
+import org.apache.geode.internal.serialization.KnownVersion;
+import org.apache.geode.logging.internal.log4j.api.LogService;
+
+public class PartitionedRegionClear {
+
+  private static final Logger logger = LogService.getLogger();
+
+  protected static final String CLEAR_OPERATION = "_clearOperation";
+
+  private final int retryTime = 2 * 60 * 1000;
```
Comment (mhansonp) [inline geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegionClear.java:44]: can this be capitalized?

**Review by mhansonp — APPROVED** [review]

**Review by jinmeiliao — APPROVED** [review]

File: geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegionClearMessage.java [hunk 3]
```diff
@@ -139,119 +136,118 @@ public int getDSFID() {
...
      * Processes this message. This method is invoked by the receiver of the message.
      *
-     * @param dm the distribution manager that is processing the message.
+     * @param distributionManager the distribution manager that is processing the message.
      */
     @Override
-    public void process(final DistributionManager dm, final ReplyProcessor21 rp) {
-      final long startTime = getTimestamp();
+    public void process(DistributionManager distributionManager,
+        ReplyProcessor21 replyProcessor21) {
```
[the remark this answers is not in the document]
Comment (kirklund) [reply geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegionClearMessage.java:238]: We've discussed this before Bill, and I disagree. Unless the Geode community discusses and agrees to change EVERY potentially final variable and parameter to be final, then there's no point. Also, final only really has an important effect on fields. If I write a method with a parameter and I never reassign that parameter to a new value, then I flat out disagree that it needs to be labeled final. I'm sorry, but if you disagree and if you feel strongly (as it seems), please bring this up in geode-dev.

[the remark this answers is not in the document]
Comment (kirklund) [reply geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegionClearMessage.java:238]: Nothing in our Design Decisions or our code standard requires final on variables or parameters. And that means I’m free to remove final from a parameter, especially if 99% of the other variables and parameters in our codebase do not have final on them. If you want to blocks these activities, then the community needs to discuss it and add it to our coding standard or our Design Decisions.

File: geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegionClearMessage.java [hunk 4]
```diff
@@ -41,96 +42,92 @@
...
-  public EventID getEventID() {
-    return eventID;
+  public PartitionedRegionClearMessage() {
+    // nothing
   }
 
-  public PartitionedRegionClearMessage() {}
+  PartitionedRegionClearMessage(Set<InternalDistributedMember> recipients,
+      PartitionedRegion partitionedRegion, ReplyProcessor21 replyProcessor21,
+      PartitionedRegionClearMessage.OperationType operationType, RegionEventImpl regionEvent) {
```
[the remark this answers is not in the document]
Comment (kirklund) [reply geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegionClearMessage.java:64]: Nothing in our Design Decisions or our code standard requires final on variables or parameters. And that means I’m free to remove final from a parameter, especially if 99% of the other variables and parameters in our codebase do not have final on them. If you want to blocks these activities, then the community needs to discuss it and add it to our coding standard or our Design Decisions.

**Comment by kirklund** [comment]
Since this PR was supposed to go to feature/GEODE-7665 and not develop, I'm closing it.

**Pull request closed without merging**